Actions button disabled instead of hidden (#434)
kendraio/kendraio-app: docs/workflow/blocks/actions.rst

        {
          "label": "Action",
          "color": "default",
          "enabledGetter": true,

>           "enabled:" true,

Reviewer: ```suggestion
          "enabled": true,
```
:sweat_smile:

Reviewer: When this is fixed it'll be good to go I think. @gsambrotta

- **buttons** (array) - the list of buttons. Each button in the list has the following properties:

  - **label** (string) - the text for the button label
  - **id** (string) - the unique identifier for the button
  - **color** (string) - passed as the “color” attribute to the material button. Use one of the supported
    Material color values, such as “primary”, “warn”, “accent”. Leave as “default” to use the default button styling for a plain button.
  - **enabledGetter** (string) - A value from the state or context. Based on this value, it generates an "enabled" property on the button.
  The button will be shown as disabled (grey out) or enabled accordingly. The default value of this property is `true`. After 
  the block is loaded, it will automatically add a property to the block json: `"enabled": true`.
  If `enabledGetter` is set to `false` the automatic property will be then shown as following: `"enabled": false`.

>   - **valueGetters** (string) - A value that will be pass as context to whichever action is being executed.

Reviewer: I think this needs to point out that it's a JMESPath mapping, that is evaluated like the Mapping block is, to determine if the button should be enabled or not. @gsambrotta

Author: @lukestanley Please suggest a text to add. I don't know much about the `valueGetters`. The text I wrote describe my knowledge of it.

.. code-block:: json
  {
    "type": "actions",
    "buttons": [
      {
        "label": "OK",
        "color": "primary",
        "blocks": [
          {
            "type": "dispatch",
            "action": "resetApp"
          }
        ]
      },
      {
        "label": "Cancel",
        "blocks": [
          {
            "type": "init"
          }
        ]
      }
    ]
  }

valueGetters
++++++++++++++++++++++
In this example, we are specifying which flow to load when the button is clicked. We do so using the `valueGetters` property and specifying an `adapter` and a `workflowId`.
You can find this example on the `Flow Cloud <https://app.kendra.io/workflowCloud/listWorkflows>`, in the list of workflows.

.. code-block:: json
  {
    "type": "actions",
    "buttons": [
        {
            "label": "Launch",
            "blocks": [
                {
                    "type": "launch",
                    "valueGetters": {
                        "adapter": "data.adapterName",
                        "workflowId": "data.id"
                    }
                }
            ],

>             "enabled": true // automatically added by the actions block. Not by a human.

Reviewer: If comments are in some JSON, it is likely to cause an error when pasted. The comment could go in the explanation text instead. @gsambrotta

Author: Ahh, you are right! Hmm, I did add the explanation in the text, but I liked to have it highlighted in here too. 
Well, I guess we cannot.

        }
    ]
}

enabledGetter
++++++++++++++++++++++

> `enabledGetter` decided if the button is enabled or disabled. This is mostly referring to the UI of the button. 

Reviewer: ```suggestion
`enabledGetter` determines if the button is enabled or disabled. This is mostly referring to the UI of the button. 
```

> In case the button is disabled, it will be displayed as grey out, and it will not be clickable.

Reviewer: ```suggestion
If the button is disabled, it will be displayed as grey out, and it will not be clickable.
```

> `enabledGetter` expect a value that will ultimately return in a `true` or `false`.

Reviewer: ```suggestion
`enabledGetter` requires a JMESPath mapping value that should return `true` or `false`.
```

The value can be added manually or taken from the state or the context.

> In this example we want to display two different button, one enabled and another disabled.

Reviewer: ```suggestion
In this example we want to display two different buttons, one enabled and another disabled.
```

To do so we take values we have saved in the state and assign it to the `enabledGetter` property:

.. code-block:: json

  {
      "type": "actions",
      "buttons": [
        {
          "label": "Visible",
          "color": "default",
          "blocks": [],
          "enabledGetter": "state.global.enabled"
        },
        {
          "label": "Disabled",
          "color": "default",
          "blocks": [],
          "enabledGetter": "state.global.disabled",
          "enabled": false
        }
      ]
    }

> After the block as loaded, the `enabled` property will be autogenerated:

Reviewer: ```suggestion
After the block as loaded, the `enabled` property will be set on the button:
```

.. code-block:: json

  {
      "type": "actions",
      "buttons": [
        {
          "label": "Visible",
          "color": "default",
          "blocks": [],
          "enabledGetter": "state.global.enabled",
          "enabled": true

        },
        {
          "label": "Disabled",
          "color": "default",
          "blocks": [],
          "enabledGetter": "state.global.disabled",
          "enabled": false
        }
      ]
    }

> Buttons are not building to be used for external links. In that case, is better to use :doc:`Template Block <template>`

Reviewer: ```suggestion
Buttons do not work for external links. Consider the :doc:`Template Block <template>` for external links.
```
